com.ibm.wala.cast.js/src/main/java/com/ibm/wala/cast/js/callgraph/fieldbased/WorklistBasedOptimisticCallgraphBuilder.java

```diff
@@ -85,62 +92,58 @@ public Set<Pair<CallVertex, FuncVertex>> extractCallGraphEdges(
         MapUtil.findOrCreateSet(reachingFunctions, fv).add(fv);
       }
     }
-
-    while (!worklist.isEmpty()) {
-      MonitorUtil.throwExceptionIfCanceled(monitor);
-
-      Vertex v = worklist.iterator().next();
-      worklist.remove(v);
-      Set<FuncVertex> vReach = MapUtil.findOrCreateSet(reachingFunctions, v);
-      for (Vertex w : Iterator2Iterable.make(flowgraph.getSucc(v))) {
+    int cnt = 0;
+    while ((bound == -1
```

Comment:
We should add a comment explaining what this loop condition is checking, as it is complex